### PR TITLE
test(ledger): Create tests for ledger chain iterator

### LINKED ISSUE #95: Create tests for ledger chain iterator
Feed in a series of block / rollback events and make sure we get the expected blocks via the chain iterator

### PR DESCRIPTION
**Pull request by arepala-uml**
1. Added a new test case TestIteratorNonInclusiveStartPoint to verify that an iterator created from a mid-chain point with inclusive=false skips the start block. 
2. Added the test case TestIteratorBlockingNextDeliversBlock to verify that Next(true) blocks when the iterator is already at the chain tip where it validates the call unblocks only after a new block is added and returns that new block.
3. Added the test case TestIteratorPostRollbackBlockDelivery to verify iterator behavior after a rollback.where it validates the iterator first returns a rollback result, then resumes by delivering blocks added after the rollback point.

Closes #95 

---
## Summary by cubic
Add three tests for the ledger `chain` iterator to validate iteration correctness and edge cases. Covers non-inclusive starts, deterministic blocking at tip, and post-rollback delivery; closes #95.

- Verifies `inclusive=false` skips the start block and begins with the next block.
- Ensures `Next(true)` blocks at tip and deterministically returns the newly added block using `github.com/blinklabs-io/dingo/internal/test/testutil` timeouts.
- Confirms after a rollback the iterator first yields the rollback, then returns blocks from the rollback point onward.

<sup>Written for commit 2f343e6d4ba9587d971aae373194728132ca2fc1. Summary will update on new commits.</sup>

### CODE REVIEW

File: chain/chain_test.go [hunk 1]
```diff
@@ -2015,3 +2016,196 @@ func TestChainReconcileEmptyForkPreservesOrphanedTip(t *testing.T) {
...
+		err error
+	}
+	ch := make(chan result, 1)
+	go func() {
+		r, err := iter.Next(true)
+		ch <- result{r, err}
+	}()
+
+	// Give the goroutine time to block, then add the next block.
+	want := testBlocks[3]
```
[the remark this answers is not in the document]
Comment (arepala-uml) [reply chain/chain_test.go:2115]: Addressed it already in my original change only and forgot to push it. I just pushed it and please validate it with my latest commit.

**Review by wolf31o2 — APPROVED** [review]
